rollup-interface/src/state_machine/stf.rs

```diff
@@ -116,19 +123,24 @@ pub trait StateTransitionFunction<Vm: Zkvm, B: BlobReaderTrait> {
     ///
     /// Applies batches of transactions to the rollup,
     /// slashing the sequencer who proposed the blob on failure.
+    /// The blobs are contained into a slot whose data is contained within the `slot_data` parameter,
+    /// this parameter is mainly used within the begin_slot hook.
     /// The concrete blob type is defined by the DA layer implementation,
     /// which is why we use a generic here instead of an associated type.
     ///
     /// Commits state changes to the database
-    fn apply_slot<'a, I>(
+    fn apply_slot<'a, I, Data: SlotData<Condition = Self::Condition>>(
         &mut self,
         witness: Self::Witness,
+        slot_data: &Data,
         blobs: I,
-    ) -> SlotResult<
-        Self::StateRoot,
-        Self::BatchReceiptContents,
-        Self::TxReceiptContents,
-        Self::Witness,
+    ) -> anyhow::Result<
```

Comment:
Can we document what does error mean in this case? Since it is quite big change to method signature.

For example, is anything stored in state? In what cases error can happen?

cc @preston-evans98

Reply:
Added a comment

Reply:
I don't see comment addressing `Result` as return type.